api/common/common_types.go

```diff
@@ -0,0 +1,120 @@
+	URL string `json:"url"`
+	// TokenEndpoint represents the API endpoint used to acquire a token (e.g., /protocol/openid-connect/token) which
+	// will be appended to the base URL to form the full URL
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OAuth Token Endpoint",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	TokenEndpoint string `json:"tokenEndpoint"`
+	// ClientSecretName represents the name of a secret (in the current namespace) which contains the client-id and
+	// client-secret values used by the OAuth client.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Client Secret",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	ClientSecretName string `json:"clientSecretName"`
+	// Scopes represents the OAuth scope values to request when acquiring a token.  Typically, this should be set to
+	// "openid" in addition to any other scopes that the SMO specifically requires (e.g., "roles", "groups", etc...) to
+	// authorize our requests
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OAuth Scopes",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	Scopes []string `json:"scopes"`
+	// UsernameClaim represents the claim contained within the OAuth JWT token which holds the username
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OAuth Username Claim",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	//+kubebuilder:default=preferred_username
+	UsernameClaim string `json:"usernameClaim"`
+	// GroupsClaim represents the claim contained within the OAuth JWT token which holds the list of groups/roles. This
+	// must be a list/array and not a space separated list of names.  It must also be a top level attribute rather than
+	// a nested field in the JSON structure of the JWT object.
+	//    i.e., {"roles": ["a", "b"]} rather than {"realm": {"roles": ["a", "b"}}.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OAuth Groups Claim",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	//+kubebuilder:default=roles
+	GroupsClaim string `json:"groupsClaim"`
+	// ClientBindingClaim represents the claim contained within the OAuth JWT token which holds the certificate SHA256
+	// fingerprint.  This is expected to be a CEL mapper expression.  It should only be changed in advanced scenarios.
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OAuth Client Binding Claim",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
+	//+kubebuilder:default="has(claims.cnf) ? claims.cnf['x5t#S256'] : []"
+	ClientBindingClaim string `json:"clientBindingClaim"`
+}
+
+// +kubebuilder:object:generate=true
+// TLSConfig defines the TLS specific attributes
+type TLSConfig struct {
+	// SecretName represents the name of a secret (in the current namespace) which contains an X.509 certificate and
```

Comment:
This doc is likely going to need to be reworded to be a little more generic and a little less ORAN specific.